xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-test/xwiki-platform-livedata-test-pageobjects/src/main/java/org/xwiki/livedata/test/po/LiveDataElement.java

```diff
@@ -95,6 +95,20 @@ public List<String> getFootnotesText()
         return getFootnotes().stream().map(WebElement::getText).collect(Collectors.toList());
     }
 
+    /**
+     * @return if the live data has finished loading and is ready for service
+     */
+    public boolean isReady()
+    {
+        // See {@link #waitUntilReady()}.
+        String[] classes = getDriver().findElementWithoutWaiting(By.id(this.id)).getAttribute("class").split("\\s+");
+        boolean isVueLoaded = !Arrays.asList(classes).contains("loading");
+        boolean areComponentsLoaded =
+            getDriver().findElement(By.id(this.id)).findElements(By.cssSelector(".xwiki-livedata .loading")).isEmpty();
+
+        return isVueLoaded && areComponentsLoaded;
```

Comment:
Wouldn't be better to add two methods, reused in both places, in order to prevent code duplication?